purldb-toolkit/README.rst

```diff
@@ -20,84 +25,541 @@ Use this command to get basic help::
 
     $ purlcli --help
     Usage: purlcli [OPTIONS] COMMAND [ARGS]...
-    
+
       Return information from a PURL.
-    
+
     Options:
       --help  Show this message and exit.
-    
+
     Commands:
       metadata  Given one or more PURLs, for each PURL, return a mapping of...
       urls      Given one or more PURLs, for each PURL, return a list of all...
-      validate  Check the syntax of one or more PURLs.
+      validate  Check the syntax and upstream repo status of one or more PURLs.
       versions  Given one or more PURLs, return a list of all known versions...
 
 
 And the following subcommands:
 
-- Validate a PURL::
+'validate' -- validate a PURL
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: none
 
     $ purlcli validate --help
     Usage: purlcli validate [OPTIONS]
-    
-      Check the syntax of one or more PURLs.
-    
+
+      Check the syntax and upstream repo status of one or more PURLs.
+
     Options:
       --purl TEXT        PackageURL or PURL.
       --output FILENAME  Write validation output as JSON to FILE.  [required]
       --file FILENAME    Read a list of PURLs from a FILE, one per line.
       --help             Show this message and exit.
 
+Examples
+########
+
+**Submit multiple PURLs using the command line:**
+
+.. code-block:: none
+
+    purlcli validate --purl pkg:npm/canonical-path@1.0.0 --purl pkg:nginx/nginx@0.8.9 --output <path/to/output.json>
+
+*Sample output:*
+
+.. code-block:: console
+
+    {
+        "headers": [
+            {
+                "tool_name": "purlcli",
+                "tool_version": "0.2.0",
+                "options": {
+                    "command": "validate",
+                    "--purl": [
+                        "pkg:npm/canonical-path@1.0.0",
+                        "pkg:nginx/nginx@0.8.9"
+                    ],
+                    "--file": null,
+                    "--output": "<path/to/output.json>"
+                },
+                "errors": [],
+                "warnings": [
+                    "'check_existence' is not supported for 'pkg:nginx/nginx@0.8.9'"
+                ]
+            }
+        ],
+        "packages": [
+            {
+                "purl": "pkg:npm/canonical-path@1.0.0",
+                "valid": true,
+                "exists": true,
+                "message": "The provided Package URL is valid, and the package exists in the upstream repo."
+            },
+            {
+                "purl": "pkg:nginx/nginx@0.8.9",
+                "valid": true,
+                "exists": null,
+                "message": "The provided PackageURL is valid, but `check_existence` is not supported for this package type."
+            }
+        ]
+    }
+
+
+**Submit multiple PURLs using a .txt file:**
+
+.. code-block:: none
+
+    purlcli validate --file <path/to/output.txt> --output <path/to/output.json>
+
+*Sample input.txt:*
+
+.. code-block:: console
+
+    pkg:npm/canonical-path@1.0.0
+    pkg:nginx/nginx@0.8.9
+
+
+Notes
```

Comment:
Maybe instead of notes, we call this `Details` and give a link to the related API page/other docs as applicable (we might not have these API pages yet, feel free to create these too, see https://github.com/nexB/purldb/issues/445#issuecomment-2140684111)

Reply:
Consider the same for all the similar sections in commands below.

Reply:
Heading changed to `Details` for all four commands.  

I've updated the `validate` sentence as you suggested.  

For the `versions` and `metadata` commands (which rely on fetchcode's `package_versions.py` and `package.py`, respectively), I removed the links to the actual code as you requested, but as far as I can tell there are no documents or similar sources to link to atm.  This is also the case for the `urls` command (which calls packageurl-python's `purl2url.py`).  As a result we currently have a `Details` link only for the `validate` command.